--- a/src/builtin/extra/builtin_xargs.c	2026-09-26 14:45:44.378014970 +0200
+++ b/src/builtin/extra/builtin_xargs.c	2026-09-26 14:45:44.378077458 +0200
@@ -1,4 +1,5 @@
 #include "../../builtin.h"
+#include "../../fdstack.h"
 #include "../../fdtable.h"
 #include "../../../lib/shell.h"
 #include "../../../lib/scan.h"
@@ -80,6 +81,10 @@
   }
 
   if(pid == 0) {
+    /* apply the shell's pending redirections/pipes to the real fds */
+    fdtable_exec();
+    fdstack_flatten();
+
     if(opts.reopen_pty) {
       int tty_fd;
 
@@ -223,7 +228,7 @@
   if(max_args != UINT_MAX && (unsigned long)args_cap > max_args)
     args_cap = max_args + 1;
 
-  struct args items;
+  struct args items = {NULL, 0};
 
   if(!(items.v = alloc(args_cap * sizeof(char*))))
     return 1;
@@ -236,12 +241,11 @@
 
     buf[r] = '\0';
 
-    if(sep == '\n') {
-      if(r > 0 && buf[r - 1] == '\n')
-        buf[--r] = '\0';
-      if(r > 0 && buf[r - 1] == '\r')
-        buf[--r] = '\0';
-    }
+    if(r > 0 && buf[r - 1] == sep)
+      buf[--r] = '\0';
+
+    if(sep == '\n' && r > 0 && buf[r - 1] == '\r')
+      buf[--r] = '\0';
 
     if(max_lines != UINT64_MAX) {
       size_t i = 0;
