diff --git a/configure.ac b/configure.ac
index 945ffc9c..98deee9e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -180,9 +180,7 @@ AC_SUBST([WIN32_TRUE],[#])
 # check for needed types
 # ---------------------------------------------------------------------------
 
-m4_define([ALL_BUILTINS],[dnl
-basename break cd dirname dump echo eval exec exit export false fdtable hash history hostname pwd set shift source test type true unset
-])dnl
+m4_define([ALL_BUILTINS],[basename, break, cd, dirname, dump, echo, eval, exec, exit, export, false, fdtable, hash, history, hostname, pwd, set, shift, source, test, type, true, unset])
 
 if test -z "${enabled_builtins}"; then
   enabled_builtins="break cd eval exec exit export hash history pwd set shift source unset"
@@ -199,27 +197,15 @@ AC_MSG_RESULT([${enabled_builtins}])
 
 uc_builtins=`echo "$enabled_builtins" | tr  '[:lower:]' '[:upper:]'`
 
-        m4_foreach([arg], [basename break cd dirname dump echo eval exec exit export false fdtable hash history hostname pwd set shift source test type true unset], [ m4_echo(m4_defn([arg]))
-        
-        ])
-
-
-AC_FOREACH(_builtin_arg_,,[
+m4_foreach([BUILTIN],[ALL_BUILTINS],[
 case "${enabled_builtins}" in
- *" _builtin_arg_" | "_builtin_arg_ "* | *" _builtin_arg_ "*) 
-
-m4_define(UC__builtin_arg_, [m4_toupper(_builtin_arg_)])
-AC_DEFINE_UNQUOTED([_builtin_arg_],[1],[Build _builtin_arg_ builtin])
+ *" BUILTIN" | "BUILTIN "* | *" BUILTIN "*)
+AC_DEFINE_UNQUOTED([BUILTIN_]m4_toupper(BUILTIN),[1],[Build the ]BUILTIN[ builtin])
   ;;
  *)
   ;;
 esac
-dnl ])
-dnl 
-dnl for builtin in ${uc_builtins}; do
-dnl 	lc_builtin=`echo "$builtin" | tr '[:upper:]' '[:lower:]'`
-dnl 	AC_DEFINE_UNQUOTED([${builtin}__builtin_arg_],[1],[Enable ${lc_builtin} builtin])
-dnl done
+])
 
 AC_SUBST([BUILTINS],[$enabled_builtins])
 
