diff --git a/src/builtin/core/builtin_chmod.c b/src/builtin/core/builtin_chmod.c
index ec28e3c7..7aacbcd6 100644
--- a/src/builtin/core/builtin_chmod.c
+++ b/src/builtin/core/builtin_chmod.c
@@ -269,7 +269,7 @@ builtin_chmod(int argc, char* argv[]) {
 
     if(chmod_only(q, "rwxXstugoa+-=,") && !chmod_only(q, "vcfR")) {
       premode = argv[c];
-      byte_copy(&argv[c], (argc - c) * sizeof(char*), &argv[c + 1]);
+      byte_copyr(&argv[c], (argc - c) * sizeof(char*), &argv[c + 1]);
       argc--;
       break;
     }
